### DISCUSSION TITLE
What is the type of Contract Instances?

### DISCUSSION THREAD

**Discussion by arcticfloyd1984** [wevm/viem#1544] [Question] [answered]
I need to set a code like this:

```
async function getContract(abi: any[], address: `0x${string}`): <ReturnType> {
      const contract = getContract({
        address: '0xFBA3912Ca04dd458c843e2EE08967fC04f3579c2',
        abi: wagmiAbi,
        publicClient,
      });
      return contract;
}
```

By hovering over the getContract I see this: 
![Screenshot 2023-11-29 at 7 00 37 PM](https://github.com/wevm/viem/assets/23353399/3e03d00c-14fc-4700-8a02-c6ba21e4d9e9)
Which is a very big type object that I can't import directly from viem. So what should be the `ReturnType` I set in my function interface?

**Comment by tmm** [comment, marked as answer]
[`GetContractReturnType`](https://github.com/wevm/viem/blob/main/src/actions/getContract.ts#L107)

**Reply by zdenham** [reply]
Are there any examples of using `GetContractReturnType` to achieve strong typing?
